pkg/controller/pravegacluster/upgrade.go

```diff
@@ -404,6 +405,178 @@ func (r *ReconcilePravegaCluster) syncSegmentStoreVersion(p *pravegav1alpha1.Pra
 	return false, nil
 }
 
+//this function is to check are we doing a rollback in case of a upgrade failure while upgrading from a version below 07 to a version above 07
+func (r *ReconcilePravegaCluster) IsClusterRollbackingFrom07(p *pravegav1alpha1.PravegaCluster) bool {
+	if !r.isRollbackTriggered(p) {
+		return false
+	}
+	//checking if sts above07 exsists
+	stsAbove07 := &appsv1.StatefulSet{}
```

Comment:
Why are we checking for existence of old/new STS to understand if rollback is triggered?
This does not look right...
I would expect this to be as simple as :
```
IsClusterRollbackingFrom07 (. . .) (. . .) {
if !r.isRollbackTriggered(p) {
		return false
	}
return true
}
```

Reply:
we need to check the existence of both sts as we need to call the method syncSegmentStoreVersionTo07()  in case of rollback only when an upgrade from a version below 07 to a version above 07 has failed meaning both sts are there and as well as rollback is triggered so i am checking the presence of both sts.

Reply:
The objective of this method is to figure out if the cluster is rolling back from 07 to a version below 07 and it should do just that and nothing else.

Reply:
I have changed it.